codemp/qcommon/cm_load.cpp

```diff
@@ -714,6 +715,31 @@ static void CM_LoadMap_Actual( const char *name, qboolean clientload, int *check
 	if ( checksum )
 		*checksum = last_checksum;
 
+	// look for a patched map
+	patchedBuf = NULL;
+	fileHandle_t p;
+	const int patchLen = FS_FOpenFileRead( va( "%s.patched", name ), &p, qfalse );
+	if ( p )
+	{
+		Com_Printf( "Found patch file %s.patched\n", name );
+		patchedBuf = Z_Malloc( patchLen, TAG_FILESYS );
```

Comment:
patch gets assigned to the old map buf, so the new memory will get freed like the old buffer would normally be w/o patch. Though I had forgotten to free the old buffer when it gets replaced, also added another free if the patch can't be read.